README.md

```diff
@@ -1,2 +1,217 @@
+
+**Initial Set-up**
+```
+docker pull bitcraze/aideck-nina
+```
+**Install IDF**
+```
+git clone https://github.com/espressif/esp-idf.git
+cd esp-idf
+git checkout v4.3.1
+git submodule sync
+git submodule update --init --recursive
+./install.sh
+```
+**Building and Flashing**
+```
+cd aideck-esp-firmware
+source ../esp-idf/export.sh
+idf.py app bootloader
+docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck-nina /bin/bash -c "/openocd-esp32/bin/openocd -f interface/ftdi/olimex-arm-usb-tiny-h.cfg -f board/esp-wroom-32.cfg -c 'program_esp32 build/bootloader/bootloader.bin 0x1000 verify' -c 'program_esp32 build/aideck_esp.bin 0x10000 verify reset exit'"
+```
+
+### 2.3 Crazyflie firmware
+
+**Configuration**
+
+```
+cd crazyflie-firmware
+make cf2_defconfig
+make menuconfig
+```
+
+In the interactive shell, select the WiFi mode. This will write a file `build/.config`.
+The WiFi credentials can be found on the TUB Cloud/imrc/network.xlsx.
+
+When successful, the `build/.config` file should contain:
+
+```
+CONFIG_DECK_AI=y
+# CONFIG_DECK_AI_WIFI_NO_SETUP is not set
+# CONFIG_DECK_AI_WIFI_SETUP_AP is not set
+CONFIG_DECK_AI_WIFI_SETUP_STA=y
+
+#
+# Credentials for access-point
+#
+CONFIG_DECK_AI_SSID="<SSID from network.xlsx>"
+CONFIG_DECK_AI_PASSWORD="<password from network.xlsx>"
+```
+
+**Building and Flashing**
+
+```
+make -j
+cfloader flash cf2.bin stm32-fw -w radio://0/80/2M
+```
+
+###  Debugging
+
+#### GAP8
+
+* Note: The output will also be shown in the CF console (e.g., with cfclient or Crazyswarm2)
+  
+* Connect CP2102 USB<->Serial adapter with Header left Pin 2 to RXD and Header left Pin 10 to GND
+* Use `dmesg` to find the device name (e.g., `/dev/ttyUSB0`)
+* Connect using cutecom (115200 baud, 8 data bits, no parity, 1 stop bit, no flow control)
+
+#### NINA
+
+* Connect CP2102 USB<->Serial adapter with Header left Pin 6 to RXD and Header left Pin 10 to GND
+* Use `dmesg` to find the device name (e.g., `/dev/ttyUSB0`)
+* Connect using cutecom (115200 baud, 8 data bits, no parity, 1 stop bit, no flow control)
+
+### 2.4 Camera Calibration
+
+* Use checkerboard (8x5, 31mm squares)
+* Make sure that the (ROS) coordinate system is: x long horizontal, y short vertical, z pointing away from front of checkerboard. Origin needs to be at the bottom-right square
+* Use `ros2 topic echo /poses` to find data for checkerboard, put it in a file named `checkerboard.yaml`
+* Use `ros2 run cvmrs_ros camera_calibration` to record data (either flying or moving manually with ROS running)
+* Get image and object points from images. Object points are in robot frame. All points are recorded into `img_obj_points.yaml` file. This also computes the actual `calibration.yaml`
+```
+python3 opencv_calibration.py PATH-TO-FOLDER
+```
+* If needed, the calibration.yaml can be re-written with different settings by calling
+```
+python3 offline_calibration.py PATH-TO-YAML
+```
```

Comment:
This is outdated. We actually just use opencv for intrinsic calibration and our optimization for the rest. There is no ROS used anymore and the checkerboard does not need to be tracked by a mocap.

Reply:
1. We record images of the checkerboard with data_collection_aideck.py
2. Then opencv_calibration_intrinsic.py -> img and obj points
3. After offline_calibration_intrinsic.py-> compute intrinsic params
4. Then extrinsic calibration part goes inside postprocess.py 

Is it correct ?

Reply:
Yes, that sounds good